### PR TITLE
[SYCLomatic] Stop process template arg when it is a parameter pack

### PR DESCRIPTION
**Pull request by yihwang-nv**
Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/test/dpct/pack_expansion_in_arg_list.cu [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+int *foo(size_t N, int Low, int High) {
+  int *Buffer = nullptr;
+  std::vector<int> Vec;
+
+  // 1. Crash in `clang::dpct::getSizeForMalloc`
+  // CHECK: DPCT1106:{{.*}}
+  CHECK_CUDA_RET(cudaMalloc<int>, &Buffer, sizeof(int) * N);
+
+  // 2. Crash in `clang::dpct::MemoryMigrationRule::memcpyMigration`
+  // CHECK: DPCT1106:{{.*}}
```
Comment (zhimingwang36) [inline clang/test/dpct/pack_expansion_in_arg_list.cu]: please add full content for 1106 in the check, easy for read and grep.

File: clang/test/dpct/pack_expansion_in_arg_list.cu [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#define CHECK_CUDA_RET(FN, ...)                                                \
+  ::detail::check_cuda_error(__FILE__, __LINE__, AS_LAMBDA(FN), __VA_ARGS__)
+
+int *foo(size_t N, int Low, int High) {
+  int *Buffer = nullptr;
+  std::vector<int> Vec;
+
+  // 1. Crash in `clang::dpct::getSizeForMalloc`
+  // CHECK: DPCT1106:{{.*}}
+  CHECK_CUDA_RET(cudaMalloc<int>, &Buffer, sizeof(int) * N);
```
Comment (zhimingwang36) [inline clang/test/dpct/pack_expansion_in_arg_list.cu:36]: Please also check  the migration result for line36.

File: clang/test/dpct/pack_expansion_in_arg_list.cu (hunk 1 above)
Comment (yihwang-nv) [reply clang/test/dpct/pack_expansion_in_arg_list.cu]: OK

Comment (yihwang-nv) [reply clang/test/dpct/pack_expansion_in_arg_list.cu]: done

File: clang/test/dpct/pack_expansion_in_arg_list.cu (hunk 2 above)
Comment (yihwang-nv) [reply clang/test/dpct/pack_expansion_in_arg_list.cu:36]: done

File: clang/test/dpct/pack_expansion_in_arg_list.cu [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#define CHECK_CUDA_RET(FN, ...)                                                \
+  ::detail::check_cuda_error(__FILE__, __LINE__, AS_LAMBDA(FN), __VA_ARGS__)
+
+int *foo(size_t N, int Low, int High) {
+  int *Buffer = nullptr;
+  std::vector<int> Vec;
+
+  // 1. Crash in `clang::dpct::getSizeForMalloc`
+  // CHECK: DPCT1106:{{.*}}: Template parameter pack expansion in function argument list is not supported.
+  // CHECK: CHECK_CUDA_RET(cudaMalloc<int>, &Buffer, sizeof(int) * N);
```
Comment (zhimingwang36) [inline clang/test/dpct/pack_expansion_in_arg_list.cu:35]: From code L37, it's diffuct to understand the real meaning of DPCT1106,  and seem it's quite specific for the currnet test case.  how about let's fix the crash issue 1st, and investigate further to fix the migration issue for the current test cases.

Comment (yihwang-nv) [reply clang/test/dpct/pack_expansion_in_arg_list.cu:35]: done

**Pull request merged**